dask/dataframe/io/parquet/arrow.py

```diff
@@ -1699,40 +1705,69 @@ def _read_table(
-
-        if use_nullable_dtypes:
-            # Determine is `pandas` or `pyarrow`-backed dtypes should be used
-            if use_nullable_dtypes == "pandas":
-                default_types_mapper = PYARROW_NULLABLE_DTYPE_MAPPING.get
+    def _determine_type_mapper(cls, **kwargs):
+        use_nullable_dtypes = kwargs.get("use_nullable_dtypes", False)
+        convert_string = kwargs.get("convert_string", False)
+        user_mapper = kwargs.get("arrow_to_pandas", {}).get("types_mapper")
+        type_mappers = []
+
+        def pyarrow_type_mapper(pyarrow_dtype):
+            # Special case pyarrow strings to use more feature complete dtype
+            # See https://github.com/pandas-dev/pandas/issues/50074
+            if pyarrow_dtype == pa.string():
+                return pd.StringDtype("pyarrow")
             else:
-                # use_nullable_dtypes == "pyarrow"
+                return pd.ArrowDtype(pyarrow_dtype)
 
-                def default_types_mapper(pyarrow_dtype):  # type: ignore
-                    # Special case pyarrow strings to use more feature complete dtype
-                    # See https://github.com/pandas-dev/pandas/issues/50074
-                    if pyarrow_dtype == pa.string():
-                        return pd.StringDtype("pyarrow")
-                    else:
-                        return pd.ArrowDtype(pyarrow_dtype)
+        # always use the user-defined mapper first, if available
+        if user_mapper is not None:
+            type_mappers.append(user_mapper)
 
-            if "types_mapper" in _kwargs:
-                # User-provided entries take priority over default_types_mapper
-                types_mapper = _kwargs["types_mapper"]
+        # next in priority is converting strings
+        if convert_string:
+            type_mappers.append({pa.string(): pd.StringDtype("pyarrow")}.get)
 
-                def _types_mapper(pa_type):
-                    return types_mapper(pa_type) or default_types_mapper(pa_type)
+        # and then nullable types
+        if use_nullable_dtypes == "pandas":
+            type_mappers.append(PYARROW_NULLABLE_DTYPE_MAPPING.get)
+        elif use_nullable_dtypes:  # "pyarrow" or True
+            type_mappers.append(pyarrow_type_mapper)
 
-                _kwargs["types_mapper"] = _types_mapper
+        def default_types_mapper(pyarrow_dtype):
+            """Try all type mappers in order, starting from the user type mapper."""
+            for type_converter in type_mappers:
+                converted_type = type_converter(pyarrow_dtype)
+                if converted_type is not None:
+                    return converted_type
 
-            else:
-                _kwargs["types_mapper"] = default_types_mapper
+        if len(type_mappers) > 0:
+            return default_types_mapper
+
+    @classmethod
+    def _arrow_table_to_pandas(
+        cls, arrow_table: pa.Table, categories, use_nullable_dtypes=False, **kwargs
+    ) -> pd.DataFrame:
```

Comment:
Can we add `convert_string=False` to the function signature here since we pass through to `_determine_type_mapper` **and** use it for index logic below?

Reply:
That would require a similar change is dask-geopandas as well, right?

Reply:
I'm not seeing why this would cause problems in dask-geopandas unless I am missing something (very possible). If it really is problematic, then feel free to ignore this suggestion